src/types/index.ts

```diff
@@ -0,0 +1,63 @@
+export interface PokemonDetails {
+  abilities: Array<Ability>;
+  base_experience: number;
+  types: Array<Type>;
+  stats: Array<Stat>;
+  sprites: Sprites;
+  height: number;
+  weight: number;
+  name: string;
+  id: number;
+}
+
+export interface PokemonData {
+  previous: null | string;
+  results: Array<Pokemon>;
+  count: number;
+  next: string;
+}
+
+export interface Pokemon {
+  name: string;
```

Comment:
or if PokemonDetails is a responce typisation.
```suggestion
  name: PokemonDetails['name'];
```

Reply:
and if you are using interface then use opportunities that it gives to you!
like 
```
export interface Pokemon extends <Pick<PokemonDetails, 'name'>> {
url: string;
}
```

Reply:
As below. I think we should not create a relation between our frontend types. Especially when PokemonDetails['name'] gaves me nothing

Reply:
acutally I changed it to type

Reply:
ok. let's leave it as is for now